src/hipscat_import/margin_cache/margin_cache_arguments.py

```diff
@@ -62,22 +63,22 @@ def _check_arguments(self):
                 raise ValueError("debug_filter_pixel_list has created empty catalog")
 
         highest_order = int(self.catalog.partition_info.get_highest_order())
-        margin_pixel_k = highest_order + 1
-        if self.margin_order > -1:
-            if self.margin_order < margin_pixel_k:
-                raise ValueError(
-                    "margin_order must be of a higher order "
-                    "than the highest order catalog partition pixel."
-                )
-        else:
-            self.margin_order = margin_pixel_k
 
-        margin_pixel_nside = hp.order2nside(self.margin_order)
+        if self.margin_order < 0:
+            self.margin_order = margin2order(margin_thr_arcmin=self.margin_threshold / 60.0)
+
+        if self.margin_order < highest_order + 1:
+            raise ValueError(
+                "margin_order must be of a higher order "
+                "than the highest order catalog partition pixel."
+            )
 
-        if hp.nside2resol(margin_pixel_nside, arcmin=True) * 60.0 < self.margin_threshold:
-            warnings.warn(
-                "Warning: margin pixels have a smaller resolution than margin_threshold; "
-                "this may lead to data loss in the margin cache."
+        margin_pixel_nside = hp.order2nside(self.margin_order)
```

Comment:
Future work: maybe just have a `order2mindist` method to call here instead, since I think that's what you really want (and could reference the new lookup values).

Reply:
Yes, it is what I should actually do